File: frontend/src/app/models/contracts.ts
export type PromptType =
  | 'zero-shot'
  | 'one-shot'
  | 'few-shot'
  | 'role-based'
  | 'json-based'
  | 'chain-of-thought';

export type OutputMode = 'text' | 'json';

export interface ModelSelection {
  id: string;
  name: string;
  symbol: string;
  provider: string;
}

export interface CompareRequest {
  prompt_type: PromptType;
  prompt_text: string;
  input_text: string;
  output_mode: OutputMode;
  role_override?: string;
  examples?: Array<{ input: string; output: string }>;
  models: ModelSelection[];
}

export interface ModelOutputResponse {
  output_id: number;
  model_name: string;
  model_symbol: string;
  provider_name: string;
  raw_response: string;
  parsed_json?: Record<string, unknown>;
  is_json_valid: boolean;
  latency_ms?: number;
}

export interface CompareResponse {
  run_id: number;
  created_at: string;
  outputs: ModelOutputResponse[];
}

export interface ModelsResponse {
  models: ModelSelection[];
}

export interface GlobalRunItem {
  run_id: number;
  created_at: string;
  prompt_type: string;
  output_mode: string;
  model_name: string;
  model_symbol: string;
  provider_name: string;
  input_text: string;
  prompt_text: string;
  raw_response: string;
  is_json_valid: boolean;
  accuracy?: number;
  clarity?: number;
  relevance?: number;
  failure_tags?: string[];
  notes?: string;
  is_saved: boolean;
}

export interface EvaluationRequest {
  output_id: number;
  accuracy: number;
  clarity: number;
  relevance: number;
  failure_tags: string[];
  notes?: string;
}

File: backend/app/schemas/contracts.py
from datetime import datetime
from typing import List, Literal, Optional

from pydantic import BaseModel, Field


class ModelSelection(BaseModel):
    id: str
    name: str
    symbol: str
    provider: str


class ExamplePair(BaseModel):
    input: str = Field(min_length=1)
    output: str = Field(min_length=1)


class CompareRequest(BaseModel):
    prompt_type: Literal["zero-shot", "one-shot", "few-shot", "role-based", "json-based", "chain-of-thought"]
    prompt_text: str = Field(min_length=1)
    input_text: str = Field(min_length=1)
    output_mode: Literal["text", "json"]
    role_override: Optional[str] = None
    examples: List[ExamplePair] = []
    models: List[ModelSelection] = Field(min_length=1)


class ModelOutputResponse(BaseModel):
    output_id: int
    model_name: str
    model_symbol: str
    provider_name: str
    raw_response: str
    parsed_json: Optional[dict] = None
    is_json_valid: bool
    latency_ms: Optional[float] = None


class CompareResponse(BaseModel):
    run_id: int
    created_at: datetime
    outputs: List[ModelOutputResponse]


class EvaluationRequest(BaseModel):
    output_id: int
    accuracy: int = Field(ge=1, le=5)
    clarity: int = Field(ge=1, le=5)
    relevance: int = Field(ge=1, le=5)
    failure_tags: List[str] = []
    notes: Optional[str] = None


class EvaluationResponse(BaseModel):
    message: str


class GlobalRunItem(BaseModel):
    run_id: int
    created_at: datetime
    prompt_type: str
    output_mode: str
    model_name: str
    model_symbol: str
    provider_name: str
    input_text: str
    prompt_text: str
    raw_response: str
    is_json_valid: bool
    accuracy: Optional[int] = None
    clarity: Optional[int] = None
    relevance: Optional[int] = None
    failure_tags: List[str] = []
    notes: Optional[str] = None
    is_saved: bool = False


class ModelsResponse(BaseModel):
    models: List[ModelSelection]
